Not generating empty structs.

ivany4 opened the pull request:
I came across the problem, where a struct that is defined after an empty one would have its values shifted in the runtime for some reason (building with Xcode 7.1). For instance, I have a code:

```
...
struct ImagesAssets {
// bunch of stuff. works well
...
} ImagesAssets;
struct Strings {
} Strings;
struct OtherAssets {
   __unsafe_unretained NSString *first; // Will have a value of second
   __unsafe_unretained NSString *second; // Will have a null value, if it's the last attribute. Otherwise will have value of third
} OtherAssets;
...
```

By eliminating empty Strings struct I managed to fix the behavior.

(I closed my original pull request and made the very same one from a different branch)

mrackwitz commented:
Thanks for your PR! I'd propose, that empty sections could be rather filtered in `#enumerate_sections` instead of passing an additional parameter. Should make the code easier to read.

ivany4 commented:
I'll have a look, when I have a chance. Could you meanwhile take a look at my branch named xcres-0.5. I added support for sounds and datasets in xcassets, started working on analysis of xibs and storyboards. And as a result, I changed the architecture quite significantly, abstracting the collection types (xcassets, bundle, loose files) from the resource types (image, sound, xib, etc). I would really like to merge it back to you at some point. But that's a big change...

mrackwitz commented:
Thanks!
Wow, this spunds super amazing! I'd love to add this features and am open to architectural improvements needed to make this happen.
Could you make a PR, so that I can review and comment on the changes?

ivany4 commented:
I did what you asked for this PR.

mrackwitz commented on lib/xcres/builder/resources_builder.rb:
```diff
@@ -177,7 +180,7 @@ def enumerate_sections
           end
         end
-        yield section_key, proc
+        yield section_key, proc, section_content.length
```
That line change seems to be not needed anymore.

ivany4 replied:
Yes, thanks for seeing it.

mrackwitz commented:
I squashed some commits together and merged that by #26. Thanks again! :+1:
